### DISCUSSION TITLE
Fraction calculation, is that possible?

### DISCUSSION THREAD

**Discussion by xuanswe** [ezylang/EvalEx#358] [Q&A] [answered]
Hi,

I successfully implemented basic features of a calculator.
Thank you a lot for the useful library!

I plan to see if I can enhance my calculator with the feature like at https://www.mathpapa.com/fraction-calculator.html

![image](https://user-images.githubusercontent.com/24372782/221348874-1b81630c-6a1e-4b5c-9434-d52fa6372084.png)

Basically, the calculator should allow to switch between decimal and fraction mode.
In fraction mode, every part, which is not an integer will be kept as original or in an optimal result.
Examples:
*  `2 / 6` should be `1 / 3`
* `sqrt(4 / 2)` should be `sqrt(2)`
*  `2 / 6 + sqrt(4 / 2)` should be `1 / 3 + sqrt(2)`

Do you see any possibility for fraction calculation with this library?

**Comment by uklimaschewski** [comment]
I moved this issue to the discussion area, as it is really not an issue.

I don't think this is easily possible with the current implementation.
Though you could do a lot of magic, when you work with the abstract syntax tree.

I think you want to do some sort of analyzing and re-arrange the expression.
The current code is only about evaluating the expression.

**Reply by xuanswe** [reply]
Does it worth to have a feature to compute the "reduced expression" in this library? Or do you think it's out of scope of this lib?

**Comment by uklimaschewski** [comment, marked as answer]
Could be an interesting feature, if somebody comes up with an implementation, I certainly would consider merging it.
